examples/gno.land/r/demo/auction/auction.gno

```diff
@@ -0,0 +1,163 @@
+	"gno.land/p/demo/mux"
+)
+
+var (
+	auctionList []*auctionp.Auction
+	currentTime time.Time
+	// router      *mux.Router
+)
+
+var router = mux.NewRouter()
+
+func init() {
+	router.HandleFunc("", renderHomepage)
+	router.HandleFunc("create/", renderCreateAuction)
+	router.HandleFunc("upcoming/", renderUpcomingAuctions)
+	router.HandleFunc("ongoing/", renderOngoingAuctions)
+	router.HandleFunc("/auction/{id}", renderAuctionDetails)
+}
+
+// CreateAuction handles the creation of a new auction
+func CreateAuction(title, description string, begin, end int64, price std.Coin) string {
+	owner := std.GetOrigCaller()
+	beginTime := time.Unix(begin, 0)
+	endTime := time.Unix(end, 0)
+	auction := auctionp.NewAuction(title, owner, description, beginTime, endTime, price)
+	auctionList = append(auctionList, auction)
+	return "Auction created successfully"
+}
+
+// PlaceBid handles placing a bid on an auction
+func PlaceBid(id int, amount std.Coin) string {
+	caller := std.GetOrigCaller()
+	if id >= len(auctionList) || id < 0 {
+		panic("Invalid auction ID")
+	}
+	auction := auctionList[id]
+	if err := auction.AddBid(caller, amount); err != nil {
+		panic(err.Error())
+	}
+	return "Bid placed successfully"
+}
+
+// EndAuction handles ending an auction
+func EndAuction(id int) string {
+	caller := std.GetOrigCaller()
+	if id >= len(auctionList) || id < 0 {
+		panic("Invalid auction ID")
+	}
+	auction := auctionList[id]
+	if err := auction.EndAuction(caller); err != nil {
+		panic(err.Error())
+	}
+	return "Auction ended successfully"
+}
+
+// Render renders the requested page
+func Render(path string) string {
+	return router.Render(path)
+}
+
+// renderHomepage renders the homepage with links to different sections
+func renderHomepage(res *mux.ResponseWriter, req *mux.Request) {
+	var b bytes.Buffer
+	b.WriteString("<h1><center>Auction Home</center></h1>\n\n")
+	b.WriteString("<h2><a href='/r/demo/auction:create/'>Create New Auction</a></h2>\n\n")
+	b.WriteString("<h2><a href='/r/demo/auction:upcoming/'>Upcoming Auctions</a></h2>\n\n")
+	b.WriteString("<h2><a href='/r/demo/auction:ongoing/'>Ongoing Auctions</a></h2>\n\n")
+	res.Write(b.String())
+}
+
+// renderCreateAuction renders the create auction page
+func renderCreateAuction(res *mux.ResponseWriter, req *mux.Request) {
+	var b bytes.Buffer
+	b.WriteString("<html><body>")
+
+	b.WriteString("#Create New Auction\n")
+	b.WriteString("<form action='/create' method='post'>\n")
+	b.WriteString("Title: <input type='text' name='title'><br>\n")
+	b.WriteString("Description: <input type='text' name='description'><br>\n")
+	b.WriteString("Begin: <input type='datetime-local' name='begin'><br>\n")
+	b.WriteString("End: <input type='datetime-local' name='end'><br>\n")
+	b.WriteString("Starting Price: <input type='number' name='price'><br>\n")
+	b.WriteString("<input type='submit' value='Create Auction'>\n")
+	b.WriteString("</form>\n")
+	b.WriteString("</body></html>")
+	res.Write(b.String())
+}
```

Comment:
This is no longer supported in gnoweb; we should rather look to have something using helplink.

Reply:
Sorry for that  🙏🏽 [da99a7a3](https://github.com/gnolang/gno/pull/2265/commits/da99a7a3a04151624e5c17b22d4433e39b08d859)